woocommerce/woocommerce-gateway-stripe
# Fix issue with Hong Kong region normalization when paying with Apple Pay

Fixes #1630 

# Changes proposed in this Pull Request:

- Instead of using the `billing_state` or `shipping_state` to normalize the region for a Hong Kong address, we instead use the `billing_postcode` and `shipping_postcode` which seem to contain the equivalent for Hong Kong addresses.

We've asked Apple whether these ^ observations about the address format are correct and are waiting for a reply. Until we've received a reply this PR will be labelled `pr: in progress`.

# Testing instructions

1. Add the following address to your Apple Pay shipping options:
    - Country: `Hong Kong`
    - Region: `New Territories`
    - District: `Taipo`
    - Address: `9 Feng Yuen Road`
2. `git checkout trunk`.
3. Add any product to your cart.
4. Attempt to checkout via Apple Pay using the address you just entered as a billing address and/or shipping address.
5. Notice that you are unable to checkout and no WooCommerce order is created.
6. `git checkout update-hong-kong-apple-pay-state-normalisation`
7. Try checking out via Apple Pay using the Hong Kong address.
8. Notice that you can now check out successfully.

-------------------
